core/tests/protractor_utils/AdminPage.js

```diff
@@ -142,6 +152,61 @@ var AdminPage = function() {
     });
   };
 
+  this.startOneOffJob = function(jobName) {
+    this._startOneOffJob(jobName, 0);
+  };
+
+  this._startOneOffJob = function(jobName, i) {
+    waitFor.visibilityOf(oneOffJobRows.first(),
+      'Starting one off jobs taking too long to appear.');
+    oneOffJobRows.get(i).getText().then((text) => {
+      if (text.toLowerCase().startsWith(jobName.toLowerCase())) {
+        oneOffJobRows.get(i).element(
+          by.css('.protractor-test-one-off-jobs-start-btn')).click();
+      } else {
+        this._startOneOffJob(jobName, ++i);
```

Comment:
This recursion seems unnecessary. Can this be done iteratively instead? Maybe by iterating over all the rows and looking for a matching job name

Reply:
The iterative solution is not the right one for this, since extracting the jobs is asynchronous so if i try an iterative approach it checks all the rows even if the matching row is found.
Suppose the job we want to start is at 7th row and there are 40 rows in total, so if we use an iterative approach like a for loop, it will execute for all the 40 rows and all the 40 rows will be checked unnecessarily. Hence recursion is very necessary here.

Reply:
Ahh, makes sense. Thanks for explaining it to me!